### ISSUE TITLE
Sonic boom of recording files.

### ISSUE THREAD

**Issue by Losses** [jwallet/spy-spotify#102]
**What is the bug ?**
> A clear and concise description of what the bug is.

[La Fille Sans Larmes.zip](https://github.com/jwallet/spy-spotify/files/3348032/La.Fille.Sans.Larmes.zip)
this audio, from 0:00 to 0:10 etc.

[Longevity.zip](https://github.com/jwallet/spy-spotify/files/3348028/Longevity.zip)
this audio, from 0:23 to 0:25 etc.

We can hear a lot of sonic boom, it appears on all audio recordings.

**How to reproduce it or what's the expected behavior ?**
> Steps to reproduce the behavior or describe the expected behavior:

Just open spotify, record.

**Track used**
> Spotify track/playlist used : https://open.spotify.com/search/recent

All tracks on my machine.

**Screenshots**
> If applicable, add screenshots to help explain your problem.
---
Want to back this issue? **[Post a bounty on it!](https://issuehunt.io/r/jwallet/spy-spotify/issues/102)** We accept bounties via [IssueHunt](https://issuehunt.io/r/jwallet/spy-spotify).

**Comment by jwallet** [comment]
you might want to check #103 to install a better driver for spytify.

**Comment by jwallet** [comment]
I was trying to debug this issue and I realized after a few tests on the same song, that I actually hear the boom while recording, so coming from Spotify.

**Comment by jwallet** [comment]
this might be fixed by 1.5

**Comment by Losses** [comment]
@jwallet I'll have a try when 1.5 released

**Comment by Losses** [comment]
@jwallet Feedback there, the problem is not solved on my machine, 1.5 installed

**Comment by jwallet** [comment]
my guess was the mp3 encoder that was not working properly, does your issue still happen when recording in wav format?

**Comment by jwallet** [comment]
@Losses I think 1.7 solved the issue, do you want to give this one a try

**Comment by Losses** [comment]
> @Losses I think 1.7 solved the issue, do you want to give this one a try

Yeah, I'll have a try while I'm free recently!

**Comment by jwallet** [comment]
closing... you can reopen it if it's still frequent.

**Comment by jwallet** [comment]
reopening, some redditors still have the issue https://www.reddit.com/r/spytify/comments/s1eyo7/recording_skipping/

**Comment by jwallet** [comment]
#92 #396 #362 might be all related, I recently found a bug on master (not released yet) I was not locking the audio device while writing the wave, resulting to another process reading it at the same time. I cannot tell if it will fix this issue since I did not have it on my PC, but it may help or even resolve some issues. 🤞

**Comment by jwallet** [comment]
https://www.reddit.com/r/spytify/comments/l9jgt8/issues_with_audio_glitchesskipping/

**Comment by adamlove86** [comment]
I'm getting skips every 2-3 songs. My guess from my end is that it might be buffer size - happens sometimes when I produce music. Is there a way to increase this?

**Comment by xource-tech** [comment]
@adamlove86 I had the same issue, from the fourth song it will happen. Your comment triggered me to try and find the issue myself.

However, when I tried the version from the master branch the problem appeared to be fixed. @jwallet the latest release is almost a year ago. Maybe publish a new version?

@adamlove86 If you're in a rush you can pull the master branch and run it yourself.

**Comment by adamlove86** [comment]
I only downloaded maybe 3, 4 months ago, so I imagine I'm on the main one?

**Comment by xource-tech** [comment]
@adamlove86 What I mean is there have been updates that are not released yet. But it is possible to run the application with those updates but it might be unstable or untested.

**Comment by adamlove86** [comment]
I would love a copy to try it out - where can I find? I've tried git clone but no executable. Do I have to build? Pls excuse my ignorance if it's obvious.

**Comment by xource-tech** [comment]
@adamlove86 Yes you need to build the app. You can do this with Visual Studio, Rider or with CLI tool

**Comment by adamlove86** [comment]
I do builds all the time with debian, but apart from downloading .net framework & Visual Studio, I have no idea how to do a Windows build. Is there the possibility of adding the instructions for perhaps cli tool?

**Comment by jwallet** [comment]
is it still fixed @xource-tech since the release of 1.11 and 1.12

**Comment by xource-tech** [comment]
@jwallet It appears so, but will check better when I have more time.
